add support for letsencrypt with public dns domain and myrtille RDP behind nginx reverse proxy on docs node

oatakan opened the pull request:
This PR makes the following changes:

- add retry for chocolatey packages installation as it was sporadically failing for me
- add additional roles, geerlingguy.epel-repo, geerlingguy-certbot, docs_nginx_ssl roles to support using letsencrypt with dns domain attached to docs node serving workstation rdp sessions via myrtille

for example https://subdomain.dns-domain-name.com/student1 would point to RDP session on Student1's workstation using real domain name and certificate so that it can survive some strict firewall environments. This will only be activated if public_dns_zone variable is defined.

- removed adding/removing gitlab and workstations to route53 as it's easier to maintain everything behind docs node and RDP session. 

- use Myrtille 2.5.5

- use urlencode for password in case special chars

cigamit commented on roles/linux-common/tasks/main.yml:
```diff
@@ -47,6 +47,9 @@
     gpgcheck: no
 
+- name: Enable optional repos
+  command: yum-config-manager --enable rhui-REGION-rhel-server-extras rhui-REGION-rhel-server-optional
```
Isn't this going to break on installs on VMWare using CentOS?

oatakan replied:
I haven't tested but good catch. it may fail on CentOS. I did put a condition for that task.

cigamit commented:
I am testing this now on EC2 and then will test it on VMWare / Azure.  I see a few other spots that I think might fail on VMWare if public_dns_zone happens to be defined.

oatakan commented:
> I am testing this now on EC2 and then will test it on VMWare / Azure. I see a few other spots that I think might fail on VMWare if public_dns_zone happens to be defined.

it turned out I introduced a typo in index.php.j2 file which I just fixed in the latest commit in this PR. It seems to work for me on EC2. I tested with public_dns_zone not defined and seems to work as expected. I will also test to see what happens when public_dns_zone is defined but not running on EC2.

oatakan commented:
> > I am testing this now on EC2 and then will test it on VMWare / Azure. I see a few other spots that I think might fail on VMWare if public_dns_zone happens to be defined.
> 
> it turned out I introduced a typo in index.php.j2 file which I just fixed in the latest commit in this PR. It seems to work for me on EC2. I tested with public_dns_zone not defined and seems to work as expected. I will also test to see what happens when public_dns_zone is defined but not running on EC2.

I tested this again and it fails when no public ip is associated in the case of VMware. I would think most commonly that would be the case. We can put conditions or leave this to user. If you're using VMware without public ip addresses don't set public_dns_zone because no implementation to handle it.
